app/src/main/kotlin/com/simplemobiletools/smsmessenger/dialogs/ScheduleSendDialog.kt

```diff
@@ -0,0 +1,154 @@
+package com.simplemobiletools.smsmessenger.dialogs
+
+import android.app.DatePickerDialog
+import android.app.DatePickerDialog.OnDateSetListener
+import android.app.TimePickerDialog
+import android.app.TimePickerDialog.OnTimeSetListener
+import android.text.format.DateFormat
+import androidx.appcompat.app.AlertDialog
+import com.simplemobiletools.commons.activities.BaseSimpleActivity
+import com.simplemobiletools.commons.extensions.*
+import com.simplemobiletools.commons.helpers.TIME_FORMAT_12
+import com.simplemobiletools.smsmessenger.R
+import com.simplemobiletools.smsmessenger.extensions.roundToClosestMultipleOf
+import com.simplemobiletools.smsmessenger.helpers.DATE_FORMAT_PATTERN
+import kotlinx.android.synthetic.main.schedule_message_dialog.view.*
+import org.joda.time.DateTime
+import java.util.*
+
+class ScheduleSendDialog(private val activity: BaseSimpleActivity, private var dateTime: DateTime? = null, private val callback: (dt: DateTime?) -> Unit) {
+    private val view = activity.layoutInflater.inflate(R.layout.schedule_message_dialog, null)
+    private val textColor = activity.getProperTextColor()
+
+    private var previewDialog: AlertDialog? = null
+    private var previewShown = false
+    private var isNewMessage = dateTime == null
+
+    private val calendar = Calendar.getInstance()
+
+    init {
+        arrayOf(view.subtitle, view.edit_time, view.edit_date).forEach { it.setTextColor(textColor) }
+        arrayOf(view.dateIcon, view.timeIcon).forEach { it.applyColorFilter(textColor) }
+        view.edit_date.setOnClickListener { showDatePicker() }
+        view.edit_time.setOnClickListener { showTimePicker() }
+        updateTexts(dateTime ?: DateTime.now().plusHours(1))
+
+        if (isNewMessage) {
+            showDatePicker()
+        } else {
+            showPreview()
+        }
+    }
+
+    private fun updateTexts(dt: DateTime) {
+        view.edit_date.text = dt.toString(DATE_FORMAT_PATTERN)
+        view.edit_time.text = dt.toString(TIME_FORMAT_12)
```

Comment:
use the selected date and time formats in the app settings everywhere, do not force any custom one

Reply:
[right?](https://github.com/SimpleMobileTools/Simple-SMS-Messenger/pull/438/commits/45c7dd77a85dc49a3cd751b1b76f3beab1500585)